executor/api/rest/server.go

```diff
@@ -144,6 +155,24 @@ func (r *SeldonRestApi) Initialise() {
 	}
 }
 
+type CloudeventHeaderMiddleware struct {
+	deploymentName string
+}
+
+func (h *CloudeventHeaderMiddleware) Middleware(next http.Handler) http.Handler {
+	// TODO: Write a test that ensures the initialise function always sets deploymentName
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		puid := r.Header.Get(payload.SeldonPUIDHeader)
+		w.Header().Set(CLOUDEVENTS_HEADER_ID_NAME, puid)
+		w.Header().Set(CLOUDEVENTS_HEADER_SPECVERSION_NAME, CLOUDEVENTS_HEADER_SPECVERSION_DEFAULT)
+		w.Header().Set(CLOUDEVENTS_HEADER_PATH_NAME, r.URL.Path)
+		w.Header().Set(CLOUDEVENTS_HEADER_TYPE_NAME, "seldon."+h.deploymentName+".response")
+		w.Header().Set(CLOUDEVENTS_HEADER_SOURCE_NAME, "seldon."+h.deploymentName)
```

Comment:
should this not be namespaced as otherwise might not be unique?

Reply:
Interesting, yeah that's a good point, I will change it so that it's namespaced

Reply:
Ok this has been namespaced and test has been added